numpy/linalg/linalg.py

```diff
@@ -2465,9 +2465,20 @@ def norm(x, ord=None, axis=None, keepdims=False):
     # Immediately handle some default, simple, fast, and common cases.
     if axis is None:
         ndim = x.ndim
+
         if ((ord is None) or
-            (ord in ('f', 'fro') and ndim == 2) or
-            (ord == 2 and ndim == 1)):
+                (ord in ('f', 'fro') and ndim == 2) or
+                (ord == 2 and ndim == 1)):
```

Comment:
I think the logic would become clearer as:
```python
# none is an alias for the default
if nd == 2 and ord in (None, 'f', 'fro'):
    fast_norm = True
elif nd == 1 and ord in (None, 2):
    fast_norm = True
elif ord is None:
    deprecate()
    fast_norm = True
else:
    fast_norm = False

if fast_norm:
    # do the fast stuff
```

Sure, this costs an extra comparison or two but I really doubt we care about that level of performance.